### PR TITLE
update otel sdk to v1.16.0/v0.39.0

### PR DESCRIPTION
**Pull request by frzifus**
## Which problem is this PR solving?
Fixes the failing CI on:
- https://github.com/jaegertracing/jaeger-operator/pull/2188
- https://github.com/jaegertracing/jaeger-operator/pull/2214
- https://github.com/jaegertracing/jaeger-operator/pull/2213
- https://github.com/jaegertracing/jaeger-operator/pull/2198

## Short description of the changes
Updating otel sdk:
https://github.com/open-telemetry/opentelemetry-go/releases/tag/v1.16.0

### CODE REVIEW

**Comment by frzifus** [comment]
hmpf.. still some stuff missing:
https://github.com/jaegertracing/jaeger-operator/actions/runs/5532303634/jobs/10094191427?pr=2261#step:6:279

**Review by iblancasa — APPROVED** [review]

File: pkg/metrics/instances.go [hunk 1]
```diff
@@ -49,10 +47,10 @@ func (i *instancesView) Record(jaeger v1.Jaeger) {
 
 func (i *instancesView) Report(ctx context.Context, observer metric.Observer) {
 	for key, count := range i.Count {
-		attrs := []attribute.KeyValue{
-			attribute.String(i.Label, key),
-		}
-		observer.ObserveInt64(i.Gauge, int64(count), attrs...)
+		opt := metric.WithAttributes(
+			attribute.Key(i.Label).String(key),
```
Comment (frzifus) [inline pkg/metrics/instances.go:51]: Let me check if that makes sense. Even if, we should rename key I guess ^^

Comment (frzifus) [reply pkg/metrics/instances.go:51]: ok, seems to be the same.

before:
```golang
// String creates a KeyValue with a STRING Value type.
func String(k, v string) KeyValue {
	return Key(k).String(v)
}
```
After:
```golang
func (k Key) String(v string) KeyValue {
	return KeyValue{
		Key:   k,
		Value: StringValue(v),
	}
}
```

**Pull request merged**